File: 超级猜图/XYSuperGuessPic-OC/XYSuperGuessPic/XYQuestion.m
//
//  XYQuestion.m
//  XYSuperGuessPic
//
//  Created by XiaoYou on 18/3/19.
//  Copyright (c) 2018年 XiaoYou. All rights reserved.
//

#import "XYQuestion.h"

@implementation XYQuestion

- (instancetype)initWithDict:(NSDictionary *)dict
{
    [self setValuesForKeysWithDictionary:dict];
    
    return self;
}

+ (instancetype)questionWithDict:(NSDictionary *)dict
{
    return [[self alloc] initWithDict:dict];
}

- (void)setValue:(id)value forUndefinedKey:(NSString *)key
{
    
}


@end

File: 超级猜图/XYSuperGuessPic/XYSuperGuessPic/XYQuestion.swift
//  代码地址：https://github.com/xiaoyouPrince/SomeUIDemos
//  Blog地址：http://xiaoyouPrince.com/
//
//  XYQuestion.swift
//  XYSuperGuessPic
//
//  Created by 渠晓友 on 2018/3/17.
//
//  Copyright © 2018年 xiayouPrince. All rights reserved.
//

import UIKit

class XYQuestion: NSObject {
    
    
    var answer : String?    ///< 答案
    var icon   : String?    ///< 对应图片
    var title  : String?    ///< 对应标题
    var options: Array<String>?     ///< 带选项
    
    
    /// 重写父类初始化方法
    override init() {
        super.init()
    }
    
    /// 自定义初始化方法
    init( dict : [String : AnyObject]) {
        super.init()
        
        /// KVC 赋值
        setValuesForKeys(dict)
        
    }
    
    /// KVC for undefined key
    override func setValue(_ value: Any?, forUndefinedKey key: String) {
        
    }
    
    

}
